assembly/pom.xml

```diff
@@ -226,5 +226,19 @@
         <parquet.deps.scope>provided</parquet.deps.scope>
       </properties>
     </profile>
+
+    <!--
+     Pull in spark-hadoop-cloud and its associated JARs,
+    -->
+    <profile>
+      <id>cloud</id>
```

Comment:
so org/apache/spark + hadoop-cloud?  I'll cause too much confusion were any JAR created thrown into a lib/ directory; you'd get 

```
hadoop-aws-2.8.1.jar
spark-core-2.3.0
hadoop-cloud-2.3.0
```
& people would be trying to understand why the hadoop-* was out of sync, who to ping, etc. 

There's actually a [hadoop-cloudproject POM](https://github.com/apache/hadoop/blob/trunk/hadoop-cloud-storage-project/hadoop-cloud-storage/pom.xml) coming in hadoop-trunk to try and be a one-stop-dependency for all cloud bindings (avoiding the ongoing "declare new dependencies per version"). the names are way too close.

I'd had it as spark-cloud, you'd felt spark-hadoop-cloud was better. I can't think of what else would do, but I do think spark- is the string which should go at the front